dashboard/app/models/course_offering.rb

```diff
@@ -60,4 +66,42 @@ def self.get_from_cache(key)
       CourseOffering.find_by_key(key)
     end
   end
+
+  def serialize
+    {
+      key: key,
+      display_name: display_name
+    }.merge(properties&.sort.to_h)
+  end
+
+  def write_serialization
+    return unless Rails.application.config.levelbuilder_mode
+    file_path = Rails.root.join("config/course_offerings/#{key}.json")
+    object_to_serialize = serialize
+    dirname = File.dirname(file_path)
+    unless File.directory?(dirname)
+      FileUtils.mkdir_p(dirname)
+    end
```

Comment:
I think you can safely remove the logic about dirname. the course_offerings directory will exist as of this PR, and it doesn't look like you are nesting deeply within it.